sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -215,22 +228,35 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
      * It will call cancelOperation if status is {@link OperationStatus#IN_PROGRESS} otherwise it does nothing.
      *
      * @throws UnsupportedOperationException when the cancel operation is not supported by the Azure service.
+     * @return A {@link Mono} containing the response.
      */
-    public void cancelOperation() throws UnsupportedOperationException {
+    public Mono<T> cancelOperation() throws UnsupportedOperationException {
```

Comment:
Anything returning a Publisher shouldn't have `throws`. It should be propagated through the error channel.

Reply:
fixed.

Reply:
This still appears to throw?